cmake/CMakeLists.txt

```diff
@@ -629,7 +639,7 @@ endif()
 #Need python to generate def file
 if(onnxruntime_BUILD_SHARED_LIB OR onnxruntime_ENABLE_PYTHON)
   if(onnxruntime_ENABLE_PYTHON)
-    if(onnxruntime_ENABLE_LANGUAGE_INTEROP_OPS)
+    if(onnxruntime_ENABLE_LANGUAGE_INTEROP_OPS OR onnxruntime_ENABLE_TRAINING_TORCH_INTEROP)
```

Comment:
I feel this is a terrible idea. We should never embed a python runtime in onnx runtime. We should never have two python runtimes in a single process.

Reply:
Talked with Changming, I will remove those unnessary link to python for those non-exe projects. Thanks @snnn.

Reply:
PR https://github.com/microsoft/onnxruntime/pull/7993 is the first step tp solve the problem.